misc/mdns/src/dns.rs

```diff
@@ -211,6 +203,37 @@ fn append_u16(out: &mut Vec<u8>, value: u16) {
     out.push((value & 0xff) as u8);
 }
 
+/// If a peer ID is longer than 63 characters, split it into segments to
+/// be compatible with RFC 1035.
+fn segment_peer_id(peer_id: &String) -> String {
```

Comment:
```suggestion
fn segment_peer_id(peer_id: String) -> String {
```

It might be premature optimization, but we could add some pass-through:

```
if peer_id.as_bytes().len() <= 63 { return peer_id }
```

Reply:
Updated 👍